### ISSUE TITLE
Validator dashboard QA

### ISSUE THREAD

**Issue by thesan** [Joystream/pioneer#4743]
## :warning: Dependencies
The QA should start only once #4742 and https://github.com/Joystream/joystream/pull/4933 get merged and `dev` is merged into validator branch one last time.

## Context

[Designs](https://www.figma.com/file/QnjZuWYtGimTEkQlN6MNlH/Validators?type=design&node-id=100-48288&mode=design&t=nWRhVkKaUSy1Vu7k-0)

Testing environment :point_right: https://dao-git-validators-joystream.vercel.app

### Dashboard itself

This should be tested at least partially on Mainnet. There has been some back and forth on how to calculate some of the values and it's important to make sure that non what is displayed is misleading to nominators. Also this page should work well in mobile:

1. Dashboard statistic headers statistic accuracy: 
  ![image](https://github.com/Joystream/pioneer/assets/6571453/fd22acc9-b35b-40fc-bc56-e39b01666ae0)
2. Dashboard list item accuracy + interaction with the filters/ordering/pagination, etc..
   ![image](https://github.com/Joystream/pioneer/assets/6571453/407f4c41-5599-46e9-8472-1a97b03e17a7)
3. Validator cards layout + data accuracy:
   ![image](https://github.com/Joystream/pioneer/assets/6571453/1e462c13-9ab7-40e0-a32d-45c298d25cfc)
4. The "Nominate" links:
   ![image](https://github.com/Joystream/pioneer/assets/6571453/88dae741-c988-4a0b-8592-8e21589c6571)
 
### Binding validator accounts to memberships

It might be easier to test this on a playground. In that case Atlas dev should be fine: https://dao-git-validators-joystream.vercel.app/#/settings?network-config=https://atlas-dev.joystream.org/network/config.json

Regardless you'll need to create validator accounts on the network used for the tests.

1. Binding a/multiple validators to a new membership.
2. Binding a/multiple validators to an existing membership.
3. Verify and unverifiy validator accounts via the CLI (https://github.com/Joystream/joystream/pull/4933) and check their statuses on the dashboard.

**Comment by ivanturlakov** [comment]
### Dashboard

Tested on https://dao-git-validators-joystream.vercel.app/ Mainnet

**✅ Dashboard statistic headers statistic accuracy**

**⚠️ Dashboard list item accuracy + interaction with the filters/ordering/pagination, etc..**
- Let us make social media icons as hyperlinks to related resources
- Search validation msg (desktop & mobile)
<img width="248" alt="Screenshot 2024-01-24 at 13 05 30" src="https://github.com/Joystream/pioneer/assets/46903215/30efad67-095d-45cf-b053-bc81681e0603">
<img width="310" alt="Screenshot 2024-01-24 at 15 32 06" src="https://github.com/Joystream/pioneer/assets/46903215/f170420d-52ba-47de-a3f2-9ff2af29229e">

**⚠️ Validator cards layout + data accuracy**
- Let us make social media icons as hyperlinks to related resources
- Customize the width of the mobile version if possible
<img width="388" alt="Screenshot 2024-01-24 at 15 34 14" src="https://github.com/Joystream/pioneer/assets/46903215/de146d94-5a98-4501-9583-02aa2ede6212">

- Nominate modal - typo “NoRminate”
<img width="461" alt="Screenshot 2024-01-25 at 13 36 57" src="https://github.com/Joystream/pioneer/assets/46903215/62f20d67-46de-40a5-973c-17fac5f51e94">

**Comment by kdembler** [comment]
No tooltip texts:
![CleanShot 2024-01-27 at 16 46 31@2x](https://github.com/Joystream/pioneer/assets/12646744/2e4d6b6a-7be7-458f-8779-f99697f10899)
![CleanShot 2024-01-27 at 16 46 25@2x](https://github.com/Joystream/pioneer/assets/12646744/3ef2761d-d15d-47a5-b3c2-c508122e525a)

Unclear "Last" in rewards section. I guess it's last era?
![CleanShot 2024-01-27 at 16 46 36@2x](https://github.com/Joystream/pioneer/assets/12646744/5dd82e09-bf89-4830-99f9-bd36995c6491)

If "All" is selected in filters, there's no text in the select:
![CleanShot 2024-01-27 at 16 46 43@2x](https://github.com/Joystream/pioneer/assets/12646744/40422fe7-bdc2-43b0-bace-705cca62248a)

Validation message shouldn't appear as I'm typing, only once I submit with Enter:

https://github.com/Joystream/pioneer/assets/12646744/168ffce3-bf87-46e4-be76-edaf4d4ba70b

When I search for something, and then remove it and confirm with Enter, the search isn't cancelled (I press Enter at the end):

https://github.com/Joystream/pioneer/assets/12646744/4ce86660-97a8-425c-a9fc-3d0d8234f8e4

**Comment by eshark9312** [comment]
@kdembler @ivanturlakov 
I fixed the issues, please check with this url.
https://dao-9pwmfuy5v-joystream.vercel.app/

**Comment by eshark9312** [comment]
@ivanturlakov 
I have not fixed the apr calculation yet. (It looks correct to me.)
Could you check again and let me know what the APRs should look?

**Comment by ivanturlakov** [comment]
Retested on https://dao-9pwmfuy5v-joystream.vercel.app/ mainnet

@eshark9312 thanks!

**Pls add a minor fix for Social links**
Incorrect Twitter link "https://web.telegram.org/k/userName"
Change Telegram link to "https://t.me/userName"

@kdembler what do you think about the formula for calculating expected Nominator APR? I am a Nominator and have checked my actual average APR and it ranges from 5 to 7% while the expected APR shows 1%

**Comment by thesan** [comment]
@ivanturlakov thank you for the quick feedback. But the correct environment is: https://dao-git-fork-eshark9312-validator-dashboard-qa-joystream.vercel.app/ (not https://dao-9pwmfuy5v-joystream.vercel.app).

On the APR thanks a lot for checking, it's very important to make sure we are not underestimating those. I'll look some more into it this week. @kdembler @eshark9312 please have look too.

**Comment by eshark9312** [comment]
@ivanturlakov 
You are right. The APR for nominators should be calculated by the following formula:
`Yearly Reward * (1 - Commission) / Stake`

**Comment by ivanturlakov** [comment]
Tested on https://37.27.43.23.nip.io/pioneer/#/settings?network-config=https://37.27.43.23.nip.io/network/config.json

### Binding a/multiple validators to a new membership

Trying to add two validators when creating a new user

⚠️ Strange behavior during validation - when adding the first validator, the Create button is inactive.

<img width="875" alt="Screenshot 2024-02-06 at 19 50 00" src="https://github.com/Joystream/pioneer/assets/46903215/ca3d3a57-af2a-441b-93ca-7fce255f450f">

 But if you add an empty field for the second validator, the button is activated.

<img width="885" alt="Screenshot 2024-02-06 at 19 50 09" src="https://github.com/Joystream/pioneer/assets/46903215/0968aebb-a618-4bdc-8227-275612fb06a5">


⚠️ I added the addresses of 2 validators but only one was bound (I suspect the first one on the list). The transaction was also performed for only one validator.

### Binding a/multiple validators to an existing membership

⚠️ Strange behavior during validation - when adding the first validator, the Save button is inactive.

<img width="879" alt="Screenshot 2024-02-06 at 20 13 01" src="https://github.com/Joystream/pioneer/assets/46903215/4ed12712-aa49-4823-8ec8-407ccd2ee0fb">

 But if you add an empty field for the second validator, the button is activated.

<img width="883" alt="Screenshot 2024-02-06 at 20 13 11" src="https://github.com/Joystream/pioneer/assets/46903215/e25edc26-52d5-4aa3-b180-d8d763367ecd">

⚠️ I added the addresses of 2 validators but only one was bound (I suspect the first one on the list). The transaction was also performed for only one validator.
<img width="883" alt="Screenshot 2024-02-06 at 19 50 23" src="https://github.com/Joystream/pioneer/assets/46903215/edc8de9b-4dd7-465f-b6a1-d2c92f4bd764">


 ⚠️ I tried to bind the second validator separately (the first one is already displayed in the profile), but when clicking Save the transaction is not initiated, but immediately shows the success window

**Comment by eshark9312** [comment]
@ivanturlakov 
After you select the validator account(or paste the address), you have to click add button, if not it is not included to the validator accounts to be bound to your membership.
And the last issue:
> I tried to bind the second validator separately (the first one is already displayed in the profile), but when clicking Save the transaction is not initiated, but immediately shows the success window

I fixed it, in this case the save button should be disabled until click the add button.

**Comment by ivanturlakov** [comment]
> After you select the validator account(or paste the address), you have to click add button, if not it is not included to the validator accounts to be bound to your membership.

<img width="821" alt="Screenshot 2024-02-08 at 10 10 49" src="https://github.com/Joystream/pioneer/assets/46903215/19686a5b-b155-4500-8363-142bf264faa3">


This is not obvious logic. Validator address fields should work like a normal form field - if there is an address there, then it is valid. The add button should only be responsible for adding a new validator address field. If we leave everything as it is, we need to add an explanation (after entering the validator, press the plus button). @kdembler @thesan what you think?

**Comment by ivanturlakov** [comment]
### Verify and unverifiy validator accounts via the CLI

Tested on https://37.27.43.23.nip.io/pioneer/#/settings?network-config=https://37.27.43.23.nip.io/network/config.json

**Works as expected**
✅ Verify **tester-6** `yarn joystream-cli working-groups:verifyValidator --group=membership 1`
✅ unVerify **tester-6** `yarn joystream-cli working-groups:verifyValidator --group=membership 1`
✅ Verify **tester-5** `yarn joystream-cli working-groups:verifyValidator --group=membership 0`



<img width="1188" alt="Screenshot 2024-02-14 at 20 36 26" src="https://github.com/Joystream/pioneer/assets/46903215/7d024f90-a413-401b-8187-14c63cb61bed">
<img width="546" alt="Screenshot 2024-02-14 at 20 50 05" src="https://github.com/Joystream/pioneer/assets/46903215/5f3fb9be-e44d-4ef7-a1f1-ccaf966071d1">
